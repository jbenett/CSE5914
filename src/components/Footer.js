import React, { Component } from "react";
import PropTypes from "prop-types";
import Button from "./Button";
import styled from "styled-components";
import DocumentManager from "./DocumentManager";

const SupportButton = styled(Button)`
    position: absolute;
    bottom: 15px;
    left: 15px;
    width: 190px;
`;

const ExitButton = styled(Button)`
    position: absolute;
    right: 0;
`;

const ManageDocumentButton = styled(Button)`
    position: absolute;
    bottom: 15px;
    width: 190px;
    left: 220px;
`;

const SaveDocumentButton = styled(Button)`
    position: absolute;
    bottom: 15px;
    left: 425px;
`;

const ModalWrapper = styled.div`
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    z-index: 1;
    overflow: auto;
    max-height: 50rem;
`;

const Header = styled.div`
  display: flex;
  justify-content: center;
  position: relative;
  width: 100%;
  padding-bottom: 10px;
`;

const ModalContent = styled.div`
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: white;
    z-index: 1;
    width: 100%;
    padding: 2rem;
    max-width: 600px;
    -webkit-box-shadow: 0px 0px 23px -1px rgba(35, 35, 35, 0.35);
    -moz-box-shadow: 0px 0px 23px -1px rgba(35, 35, 35, 0.35);
    box-shadow: 0px 0px 23px -1px rgba(35, 35, 35, 0.35);
    border-radius: 3px;
`;

const ModalOverlay = styled.div`
    background-color: rgba(255, 255, 255, 0.75);
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    z-index: 0;
`;

class Footer extends Component {
    static propTypes = {
        onChangeDocument: PropTypes.func,
        onSaveDocument: PropTypes.func
    };

    state = {
        showDocuments: false
    };

    _onChangeDocument = doc => {
        const { onChangeDocument } = this.props;
        onChangeDocument(doc);
        this.setState({ showDocuments: false });
    };

    render() {
        const { showDocuments } = this.state;
        const { onSaveDocument } = this.props;

        return (
            <div>
                <SupportButton>🤑&nbsp;Support The Boys&nbsp;🤑</SupportButton>
                <ManageDocumentButton
                    onClick={() =>
                        this.setState({ showDocuments: !showDocuments })
                    }
                >
                    📝&nbsp;Manage Documents
                </ManageDocumentButton>
                <SaveDocumentButton onClick={onSaveDocument}>
                    💾&nbsp;Save Document
                </SaveDocumentButton>
                {showDocuments && (
                    <ModalWrapper>
                        <ModalOverlay />
                        <ModalContent>
                          <Header>
                            <h3>Saved Documents</h3>
                            <ExitButton onClick={() =>
                                this.setState({ showDocuments: !showDocuments })
                            }> Close </ExitButton>
                          </Header>
                            <DocumentManager
                                onChangeDocument={this._onChangeDocument}
                            />
                        </ModalContent>
                    </ModalWrapper>
                )}
            </div>
        );
    }
}

export default Footer;
