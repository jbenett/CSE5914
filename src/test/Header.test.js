import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../components/Header.js'

describe('Header', () => {
  it('should render correctly', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
