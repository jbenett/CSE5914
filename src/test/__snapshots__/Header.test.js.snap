// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef
        id="logo"
        src="logo.PNG"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "id": "logo",
        "src": "logo.PNG",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Group": [Function],
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  position: absolute;
  left: 1rem;
  top: 1rem;
  background-color: white;
  border-radius: 3px;
  height: 6rem;
  width: 6rem;
  ",
            [Function],
            "
",
          ],
        },
        "create": [Function],
        "displayName": "Styled(Image)",
        "foldedComponentIds": Array [],
        "handledProps": Array [
          "as",
          "avatar",
          "bordered",
          "centered",
          "children",
          "circular",
          "className",
          "content",
          "dimmer",
          "disabled",
          "floated",
          "fluid",
          "hidden",
          "href",
          "inline",
          "label",
          "rounded",
          "size",
          "spaced",
          "ui",
          "verticalAlign",
          "wrapped",
        ],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          id="logo"
          src="logo.PNG"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "logo",
          "src": "logo.PNG",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Group": [Function],
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
  position: absolute;
  left: 1rem;
  top: 1rem;
  background-color: white;
  border-radius: 3px;
  height: 6rem;
  width: 6rem;
  ",
              [Function],
              "
",
            ],
          },
          "create": [Function],
          "displayName": "Styled(Image)",
          "foldedComponentIds": Array [],
          "handledProps": Array [
            "as",
            "avatar",
            "bordered",
            "centered",
            "children",
            "circular",
            "className",
            "content",
            "dimmer",
            "disabled",
            "floated",
            "fluid",
            "hidden",
            "href",
            "inline",
            "label",
            "rounded",
            "size",
            "spaced",
            "ui",
            "verticalAlign",
            "wrapped",
          ],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
